metals/src/main/scala/scala/meta/internal/metals/codeactions/ExtractRenameMember.scala

```diff
@@ -378,20 +424,29 @@ class ExtractRenameMember(
     def removeTreeEdits(t: Tree): List[l.TextEdit] =
       List(new l.TextEdit(t.pos.toLSP, ""))
 
-    val packageEdit = member.parent
+    val packageEdit = endableMember.member.parent
       .flatMap {
         case p: Pkg
             if p.stats.forall(t =>
-              t.isInstanceOf[Import] || t.equals(member) || companion
-                .exists(_.equals(t))
+              t.isInstanceOf[Import] || t
+                .equals(endableMember.member) || maybeEndableMemberCompanion
+                .exists(_.member.equals(t))
             ) =>
           Some(p)
         case _ => None
       }
       .map(removeTreeEdits)
 
     packageEdit.getOrElse(
-      removeTreeEdits(member) ++ companion.map(removeTreeEdits).getOrElse(Nil)
+      removeTreeEdits(endableMember.member) ++ maybeEndableMemberCompanion
+        .map(_.member)
+        .map(removeTreeEdits)
+        .getOrElse(Nil) ++ endableMember.maybeEndMarker
+        .map(removeTreeEdits)
+        .getOrElse(Nil) ++ maybeEndableMemberCompanion
+        .flatMap(_.maybeEndMarker)
+        .map(removeTreeEdits)
+        .getOrElse(Nil)
```

Comment:
```suggestion
      removeTreeEdits(endableMember.member) ++ { 
        maybeEndableMemberCompanion
        .map(_.member) ++ endableMember.maybeEndMarker
        ++ maybeEndableMemberCompanion
        .flatMap(_.maybeEndMarker)
        }.map(removeTreeEdits)
```
maybe something along these lines?

Reply:
I mean I can define a separate `List[Tree]` to make it work! but would it look prettier?

Reply:
I think it would be good in general to make it a bit more readable.

Reply:
Done